src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -599,6 +620,42 @@ internal void ValidateCanBeUsedForPropertyMetadataSerialization()
 
         private IJsonTypeInfoResolver? _originatingResolver;
 
+        /// <summary>
+        /// Gets or sets an attribute provider corresponding to the deserialization constructor.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        /// The <see cref="JsonPropertyInfo"/> instance has been locked for further modification.
+        /// </exception>
+        /// <remarks>
+        /// When resolving metadata via the built-in resolvers this will be populated with
+        /// the underlying <see cref="ConstructorInfo" /> of the serialized property or field.
+        /// </remarks>
+        public ICustomAttributeProvider? ConstructorAttributeProvider
+        {
+            get
+            {
+                ICustomAttributeProvider? ctorAttrProvider = _constructorAttributeProvider;
+                if (ctorAttrProvider is null && ConstructorAttributeProviderFactory is { } factory)
+                {
+                    _constructorAttributeProvider = ctorAttrProvider = factory();
```

Comment:
If the factory returns null, it'll be reinvoked on each access. Is that ok?

Reply:
No. That should have been clearing the factory property on initialization. I'll pushed an update.